wire-library/wire-schema/src/commonMain/kotlin/com/squareup/wire/schema/Linker.kt

```diff
@@ -135,15 +143,25 @@ class Linker {
         continue
       }
 
+      if (loadExhaustively) {
```

Comment:
I think we can skip the entire retain() step completely if loadExhaustively is used

Reply:
ie. on line 141 do, ` if (loadExhaustively || sourceFiles.contains(fileLinker))`

Reply:
Fixed!